api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsService.java

```diff
@@ -370,16 +370,19 @@ public boolean areUserCreditsExpired(DbUser user) {
 
   // Returns true if the user's credits are expiring within the initialCreditsExpirationWarningDays.
   private boolean areCreditsExpiringSoon(DbUser user) {
-    long initialCreditsExpirationWarningDays =
+    List<Long> initialCreditsExpirationWarningDays =
         workbenchConfigProvider.get().billing.initialCreditsExpirationWarningDays;
     return getCreditsExpiration(user)
         .map(
-            expirationTime ->
-                clockNow()
-                    .after(
-                        new Timestamp(
-                            expirationTime.getTime()
-                                - TimeUnit.DAYS.toMillis(initialCreditsExpirationWarningDays))))
+            expirationTime -> {
+              long expirationMillis = expirationTime.getTime();
+              Timestamp now = clockNow();
+              return initialCreditsExpirationWarningDays.stream()
+                  .anyMatch(
+                      days ->
+                          now.after(
+                              new Timestamp(expirationMillis - TimeUnit.DAYS.toMillis(days))));
+            })
         .orElse(false);
   }
```

Comment:
This and the change on lines 474 - 478 will have the added change of letting initial credits be extended up to 90 days before they expire. That may be fine, but it would be good to confirm with Moira that we want to allow that.